tests/system/model_monitoring/test_model_monitoring.py

```diff
@@ -573,6 +574,44 @@ def test_mep_with_model(self):
         assert mep_3.spec.feature_names == ["f1"]
         assert mep_3.spec.label_names == ["l1", "l2"]
 
+    def test_mep_with_model_runner(self):
+        function = mlrun.code_to_function(
+            name="function_with_model",
+            kind="serving",
+            tag="latest",
+            project=self.project_name,
+            filename=str(self.assets_path / "models.py"),
+            image=self.image,
+        )
+        function.save(versioned=False)
+        graph = function.set_topology("flow", engine="async")
+        model_runner_step = mlrun.serving.states.ModelRunnerStep(name="model-runner")
+        model_runner_step.add_model(
+            model_class="IncModel", endpoint_name="my-model-1", inc=1
+        )
+        model_runner_step.add_model(
+            model_class="IncModel", endpoint_name="my-model-2", inc=2
+        )
+        graph.to(model_runner_step, "runner").respond()
```

Comment:
Please add a small step before that, maybe even a step that does nothing, just to verify that the root-finding iteration really works (it would catch the bug you currently have there).